src/common/transformations/src/transformations/sdpa_to_paged_attention/position_ids_replacer.cpp

```diff
@@ -123,3 +125,30 @@ ov::pass::PositionIDsReplacerQwen::PositionIDsReplacerQwen(const Output<Node>& p
     auto m = std::make_shared<Matcher>(p_slice_2, matcher_name);
     register_matcher(m, callback);
 }
+
+ov::pass::PositionIDsReplacerCodeGen2::PositionIDsReplacerCodeGen2(
+    const std::shared_ptr<ov::op::v0::Parameter>& position_ids) {
+    MATCHER_SCOPE(PositionIDsReplacerCodeGen2);
+
+    auto p_reshape = ov::pass::pattern::wrap_type<v1::Reshape>();
+    auto p_unsq = ov::pass::pattern::wrap_type<v0::Unsqueeze>();
+    auto p_add = ov::pass::pattern::wrap_type<v1::Add>();
+    auto p_slice = ov::pass::pattern::wrap_type<v8::Slice>({p_unsq, p_reshape, p_add, any_input(), any_input()});
```

Comment:
this is not a so specific pattern
need to hardcode more operations to make it unique for CodeGen2